src/main.rs

```diff
@@ -7,35 +7,56 @@ use nalgebra;
 mod rendering;
 
 use std::io::BufRead;
+use std::time::Instant;
 use winit::{
     event::{Event, WindowEvent},
     event_loop::{ControlFlow, EventLoop},
     window::WindowBuilder,
 };
 
+use imgui::*;
+use imgui_winit_support;
+
 enum MouseMode {
     Cursor,
     CameraLook,
     CameraPan,
 }
 
+struct UiState {
+    camera_target: [f32; 3],
+    camera_range: f32,
+    camera_azimuth: f32,
+    camera_elevation: f32,
+    gui_open: bool,
+}
```

Comment:
Types like these could probably go in their own files specific to the UI processing.